test/ng2-bs3-modal.spec.ts

```diff
@@ -1,124 +1,142 @@
-import {
-    beforeEach,
-    afterEach,
-    ddescribe,
-    xdescribe,
-    describe,
-    expect,
-    iit,
-    inject,
-    beforeEachProviders,
-    it,
-    xit
-} from '@angular/core/testing';
-import { ComponentFixture, TestComponentBuilder } from '@angular/compiler/testing';
-import { SpyLocation } from '@angular/common/testing';
-import { Component, ComponentResolver, ViewChild, ContentChild, provide, OnDestroy, Input } from '@angular/core';
+import { APP_BASE_HREF } from '@angular/common';
+import { async, inject, addProviders, ComponentFixture, TestComponentBuilder } from '@angular/core/testing';
+import { Injectable, Input, Injector, Component, ComponentResolver, ViewChild } from '@angular/core';
 import { Location } from '@angular/common';
-import { Router, Routes, ROUTER_DIRECTIVES, RouterOutletMap, RouterUrlSerializer, DefaultRouterUrlSerializer } from '@angular/router';
+import { Router, ROUTER_DIRECTIVES, RouterOutletMap, provideRouter, RouterConfig, DefaultUrlSerializer, UrlSerializer, ActivatedRoute } from '@angular/router';
+import { SpyLocation } from '@angular/common/testing';
+
 import { ModalComponent, MODAL_DIRECTIVES } from '../src/ng2-bs3-modal/ng2-bs3-modal';
 
-// Needed because ViewChild isn't resolved anymore in the new router
-// https://github.com/angular/angular/issues/4452
-class GlueService {
-    testComponent: TestComponent;
+
+@Component({
```

Comment:
Created components up here since the one test that uses routing needed a reference to a parent component. It was not working otherwise